tests/components/tado/test_config_flow.py

```diff
@@ -22,7 +23,71 @@ def _get_mock_tado_api(getMe=None):
     return mock_tado
 
 
-async def test_form(hass: HomeAssistant) -> None:
+@pytest.mark.parametrize(
+    ("exception", "error"),
+    [
+        (KeyError, "invalid_auth"),
+        (RuntimeError, "cannot_connect"),
+        (ValueError, "unknown"),
+    ],
+)
+async def test_form_exceptions(
+    hass: HomeAssistant, exception: Exception, error: str
+) -> None:
+    """Test we handle Exception."""
+    result = await hass.config_entries.flow.async_init(
+        DOMAIN, context={"source": config_entries.SOURCE_USER}
+    )
+
+    with patch(
+        "homeassistant.components.tado.config_flow.Tado",
+        side_effect=exception,
+    ):
+        result2 = await hass.config_entries.flow.async_configure(
+            result["flow_id"],
+            {"username": "test-username", "password": "test-password"},
+        )
+
+    assert result2["type"] == "form"
+    assert result2["errors"] == {"base": error}
+
+
+async def test_options_flow(hass: HomeAssistant) -> None:
+    """Test config flow options."""
+    entry = MockConfigEntry(domain=DOMAIN, data={"username": "test-username"})
+    entry.add_to_hass(hass)
+
+    result = await hass.config_entries.flow.async_init(
+        DOMAIN, context={"source": config_entries.SOURCE_USER}
+    )
+    assert result["type"] == "form"
+    assert result["errors"] == {}
+
+    with patch(
+        "homeassistant.components.tado.async_setup_entry",
+        return_value=True,
+    ):
+        await hass.config_entries.async_setup(entry.entry_id)
+        await hass.async_block_till_done()
+
+    result = await hass.config_entries.options.async_init(
+        entry.entry_id, context={"source": config_entries.SOURCE_USER}
+    )
+    assert result["type"] == "form"
+    assert result["step_id"] == "init"
+
+    result2 = await hass.config_entries.options.async_configure(
+        result["flow_id"],
+        {"fallback": "TADO_DEFAULT"},
```

Comment:
This looks great! Could you maybe still use the constants? CONF_FALLBACK and that default one? :)

Reply:
Sure, let's pollish this up! ;)